src/parser/ast/mod.rs

```diff
@@ -17,11 +17,25 @@ pub struct DebugSymRef {
     pub end: usize,
     /// Source file reference
     file: Arc<SimpleFile<String, String>>,
+    /// Created by the compiler
+    pub virt: bool,
 }
 
 impl DebugSymRef {
     pub fn new(start: usize, end: usize, file: Arc<SimpleFile<String, String>>) -> DebugSymRef {
-        DebugSymRef { start, end, file }
+        DebugSymRef {
+            start,
+            end,
+            file,
+            virt: false,
+        }
+    }
+
+    pub fn virt(other: &DebugSymRef) -> DebugSymRef {
```

Comment:
At first I was confused with the function (also given the argument name "other"), I think it could be named `make_virtual` or `into_virtual`, and "other" can be just "dsym" since this is not an instance method.